### PR TITLE
Component/joblist

### PR DESCRIPTION
**Pull request by RemyaManoharan**
## Ticket Link: [ID[](https://www.meistertask.com/app/task/ID](https://www.meistertask.com/app/task/7vF4BMb0/create-components-for-list-of-jobs)) https://www.meistertask.com/app/task/b9gort87/create-a-component-for-job-details-page
https://www.meistertask.com/app/task/1f6d01Ew/added-filter-form-in-home-page
## 📝 Description

- [x] Description of work : Added components for JobList, JobCard, Jobdetail, Apply Job , in which Apply Job is in progress. Also added filter form in home page where structure is only completed. need to add functionality for filtering. SO jobList, JobCard and Job detail completed.. need some more stylling in all files.

### CODE REVIEW

File: frontend/src/components/ApplyForm/ApplyForm.tsx
Comment (BrantalikP) [inline frontend/src/components/ApplyForm/ApplyForm.tsx]: Is this file used somewhere?

File: frontend/src/components/FilterForm/FilterForm.tsx
Comment (BrantalikP) [inline frontend/src/components/FilterForm/FilterForm.tsx]: delete unused code

File: frontend/src/components/FilterForm/FilterForm.tsx
Comment (BrantalikP) [inline frontend/src/components/FilterForm/FilterForm.tsx:2]: Nice Work! just please check the designs one more time and try to fix the design compliance, also please check the web responsivity. 🙏
Screenshot 2023-10-01 at 12 38 58

**Review by BrantalikP — CHANGES_REQUESTED** [review]
Hey @RemyaManoharan, first of all, great job! 👏 You've put in a lot of work into this PR and I also appreciate your detailed descriptions and attaching the task link. 🙏
However, I noticed that this PR is quite large and includes a lot of unused code (e.g., ﻿Filter from ﻿`ApplyForm`). As a result, I'm unsure of which parts specifically to review. Could you please remove any unused code and files? For future reference, it would be helpful to break down the PRs into smaller, more manageable parts. Instead of including all components at once, you could create separate PRs (one for ﻿Filter, another for ﻿Job Detail, one for the DB, and so on). This approach would significantly improve the review process.

File: frontend/src/components/FilterForm/FilterForm.tsx
Comment (RemyaManoharan) [reply frontend/src/components/FilterForm/FilterForm.tsx:2]: Thanks Petr for the review..I will do responsiveness and apply form  in the current week as a new task in meister.. For apply form I did some structuring with dummy data as I don't want to waste time till routes are ready to shared.It would be nice if I can share my code with teams so that they can also work on the filter component , apply form and styling.
The files to be reviewed are JobList , JobCard, JobDetail.. Fixing the styling for responsiveness will be doing this week.

Comment (BrantalikP) [reply frontend/src/components/FilterForm/FilterForm.tsx:2]: Hey @RemyaManoharan! ok so if I understand it correctly, you are supposed to do only the "skeleton" of the components, right? If so, please create a ticket for each component that needs the stylings fix and responsivity before we can proceed with the review so I know that the work will be done :) Thank you 🙏

Also some small things:
1. I noticed that you shared some code with @sayali05 's PR, can you remove everything that is not related to the components? I reviewed the Backend work as a part of this [PR](https://github.com/HackYourFuture-CPH/class-25-ironside/pull/17) already. Contrary we can wait until the @sayali05 is merged first

2. Please check and resolve the lining warnings 🙏 You can see them if you click on "files changed" tab here on github. It should be just simple removing unused code. 
Screenshot 2023-10-02 at 9 20 37

3. I noticed that fonts are off, will we add them in a separate PR?

Thank you and GJ! ❤️

Comment (RemyaManoharan) [reply frontend/src/components/FilterForm/FilterForm.tsx:2]: Thanks Petr.. I will be resolve lining warning and pushing again.. I need Sayali's backend routes to fetch jobs using jobstore.tsx and to render the jobs using JobCards and JobDetail.. So If I delete the Sayalis code I will be getting errors.

File: frontend/src/pages/home/Home.css
Comment (BrantalikP) [inline frontend/src/pages/home/Home.css]: remove comment please

File: frontend/src/components/JobList/JobList.css
Comment (BrantalikP) [inline frontend/src/components/JobList/JobList.css]: unused code

File: frontend/src/components/JobDetail/JobDetail.tsx
Comment (BrantalikP) [inline frontend/src/components/JobDetail/JobDetail.tsx]: remove the commented code please :)

File: frontend/src/components/FilterForm/FilterForm.tsx
Comment (BrantalikP) [reply frontend/src/components/FilterForm/FilterForm.tsx:2]: Oki @RemyaManoharan , understand.. Then please resolve all comments and linting errors, and I will make sure that there are no conflicts after that we can merge it is it ok ?:)

also please share the newly created tickets for each component that needs the new styling here so I can refer to it in the future 🙏. Could you create a ticket for adding correct fonts for me, please? Thank you! ❤️

**Review by BrantalikP — CHANGES_REQUESTED** [review]

File: frontend/src/components/FilterForm/FilterForm.tsx
Comment (RemyaManoharan) [reply frontend/src/components/FilterForm/FilterForm.tsx:2]: Ok.. I will be doing all the mentioned and will be pushing again.. Thanks :)

**Comment by RemyaManoharan** [comment]
https://www.meistertask.com/app/task/gVZeDHEi/make-jobcard-and-filter-component-responsive
https://www.meistertask.com/app/task/bHU0NJmg/remove-acces-token-from-joblist-when-it-will-be-dynamic
https://www.meistertask.com/app/task/m5Uzn5Zh/making-filter-component-functional-as-now-only-structure-is-completed
These are the created tickets to be done for styling.

**Review by BrantalikP — APPROVED** [review]
👏

**Pull request merged**
